core/chains/evm/bulletprooftxmanager/transmitchecker.go

```diff
@@ -0,0 +1,160 @@
+package bulletprooftxmanager
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/vrf_coordinator_v2"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var (
+	_ TransmitCheckerFactory = &CheckerFactory{}
+	_ TransmitChecker        = NoChecker{}
+	_ TransmitChecker        = &SimulateChecker{}
+	_ TransmitChecker        = VRFV2Checker{}
+)
+
+// CheckerFactory is a real implementation of TransmitCheckerFactory.
+type CheckerFactory struct {
+	Client evmclient.Client
+}
+
+// BuildChecker satisfies the TransmitCheckerFactory interface.
+func (c *CheckerFactory) BuildChecker(spec TransmitCheckerSpec) (TransmitChecker, error) {
+	switch spec.CheckerType {
+	case TransmitCheckerTypeSimulate:
+		return &SimulateChecker{c.Client}, nil
+	case TransmitCheckerTypeVRFV2:
+		coord, err := vrf_coordinator_v2.NewVRFCoordinatorV2(spec.VRFCoordinatorAddress, c.Client)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create coordinator at address %v", spec.VRFCoordinatorAddress)
```

Comment:
Nit: information about VRF version would be useful in the error msg: "failed to create coordinator v2 at address %v"

Reply:
Good call, done